speasy/webservices/amda/_impl.py

```diff
@@ -84,15 +85,46 @@ def update_inventory(self):
         data_tree.amda.Parameters.__dict__.update(data.dataRoot.AMDA.__dict__)
 
         self._update_lists()
+    def parameter_concat(self, param1, param2):
+        """Concatenate parameters
+        """
+        if param1 is None and param2 is None:
+            return None
+        if param1 is None:
+            return param2
+        if param2 is None:
+            return param1
+        param1.time = np.hstack((param1.time, param2.time))
+        if len(param1.data.shape) == 1:
+            param1.data = np.hstack((param1.data, param2.data))
+        else:
+            param1.data = np.vstack((param1.data, param2.data))
+        return param1
 
     def dl_parameter(self, start_time: datetime, stop_time: datetime, parameter_id: str, **kwargs) -> Optional[
         SpeasyVariable]:
-
-        start_time = start_time.timestamp()
-        stop_time = stop_time.timestamp()
+        if isinstance(start_time, datetime):
+            start_time = start_time.timestamp()
+        if isinstance(stop_time, datetime):
+            stop_time = stop_time.timestamp()
+
+        dt = timedelta(days=1).total_seconds()
+
+        if stop_time - start_time > dt:
```

Comment:
If we follow this solution, I think this should rather be split in two methods:
-> dl_parameter which now does some sort of request split if more than max_dt is asked
-> dl_parameter_chunk which basically do what dl_parameter used to do

Like this we avoid recursive call with complex logic and abstraction level mix-up

Reply:
done